### PR TITLE
Updated plugin tutorial w/ dialog & dep warning.

### LINKED ISSUE #2732: Document the plugin creation dialog
The editor has a GUI dialog that was added a while ago which creates all files neccessary for an editor plugin. The editor plugin tutorials should be updated to mention this dialog.

### PR DESCRIPTION
**Pull request by willnationsdev**
Closes #2732.

- New note with CreatePluginDialog and images.
- Warns users that CustomTypes will be deprecated in 4.0.

### CODE REVIEW

**Comment by skyace65** [comment]
I don't think this is a good way to update the page. I think the instructions to create the script files manually should be replaced with instructions on how to use the create plugin dialog. There's no reason to do things manually instead of using the dialog.

**Comment by willnationsdev** [comment]
@skyace65 I have updated it to be a more in-depth replacement now.

**Comment by skyace65** [comment]
Everything looks good to me.

**Comment by willnationsdev** [comment]
Ping. For whoever is ready to merge this.

File: tutorials/plugins/editor/making_plugins.rst [hunk 1]
```diff
@@ -128,6 +126,15 @@ sometimes it can be cumbersome, especially if you're using it in many
 projects. A good solution to this is to make a plugin that adds a node with a
 custom behavior.
 
+.. warning::
+
+  Nodes added via an EditorPlugin are "CustomType" nodes. While they work
+  with any scripting language, they have fewer features than
+  :ref:`the Script Class system <doc_scripting_continued_class_name>`. If you
+  are writing GDScript or NativeScript, we recommend using them instead.
+  Godot 4.0 will have full Script Class support for all languages. At that
```
Comment (akien-mga) [inline tutorials/plugins/editor/making_plugins.rst]: I would not mention this, as it's not written in stone and we can't promise it will happen.

**Comment by akien-mga** [comment]
Thanks!

**Pull request merged**